jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/internal/HttpChannelState.java

```diff
@@ -921,6 +870,16 @@ public long getTimeStamp()
             return _timeStamp;
         }
 
+        @Override
+        public long getNanoTimeStamp()
+        {
+            HttpStream stream = _httpChannel.getHttpStream();
+            if (stream != null)
+                return stream.getNanoTimeStamp();
+
+            return NanoTime.now() - TimeUnit.MILLISECONDS.toNanos(System.currentTimeMillis() - getTimeStamp());
```

Comment:
Should call `super` instead?
As we don't want to duplicate the (wrong) logic here.

Reply:
Does super call a default implementation? I don't think so.